src/course-unit/breadcrumbs/Breadcrumbs.test.jsx

```diff
@@ -0,0 +1,112 @@
+import MockAdapter from 'axios-mock-adapter';
+import { render, waitFor, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { initializeMockApp } from '@edx/frontend-platform';
+import { AppProvider } from '@edx/frontend-platform/react';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+
+import { getCourseOutlineIndexApiUrl } from '../../course-outline/data/api';
+import { fetchCourseOutlineIndexQuery } from '../../course-outline/data/thunk';
+import { courseOutlineIndexMock } from '../../course-outline/__mocks__';
+import initializeStore from '../../store';
+import { executeThunk } from '../../utils';
+import { getCourseUnitApiUrl } from '../data/api';
+import { fetchCourseUnitQuery } from '../data/thunk';
+import { courseUnitIndexMock } from '../__mocks__';
+import Breadcrumbs from './Breadcrumbs';
+
+import messages from './messages';
+
+let axiosMock;
+let store;
+const courseId = '123';
+const breadcrumbsExpected = {
+  section: {
+    id: 'block-v1:edX+DemoX+Demo_Course+type@chapter+block@interactive_demonstrations',
+    displayName: 'Example Week 1: Getting Started',
+  },
+  subsection: {
+    displayName: 'Lesson 1 - Getting Started',
+  },
+};
+
+const renderComponent = () => render(
+  <AppProvider store={store}>
+    <IntlProvider locale="en">
+      <Breadcrumbs courseId={courseId} />
+    </IntlProvider>
+  </AppProvider>,
+);
+
+describe('<Breadcrumbs />', () => {
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+
+    store = initializeStore();
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+    axiosMock
+      .onGet(getCourseUnitApiUrl(courseId))
+      .reply(200, courseUnitIndexMock);
+    await executeThunk(fetchCourseUnitQuery(courseId), store.dispatch);
+  });
+
+  it('render Breadcrumbs component correctly', async () => {
+    const { getByText } = renderComponent();
+
+    await waitFor(() => {
+      expect(getByText(breadcrumbsExpected.section.displayName)).toBeInTheDocument();
+      expect(getByText(breadcrumbsExpected.subsection.displayName)).toBeInTheDocument();
+    });
+  });
+
+  it('render dropdown loading placeholder on pending', async () => {
+    const { getByText, queryAllByTestId } = renderComponent();
+
+    await waitFor(() => {
+      expect(getByText(breadcrumbsExpected.section.displayName)).toBeInTheDocument();
+      expect(getByText(breadcrumbsExpected.subsection.displayName)).toBeInTheDocument();
+    });
+
+    const button = getByText(breadcrumbsExpected.section.displayName);
+    await act(async () => {
+      userEvent.click(button);
```

Comment:
[nit] [userEvent](https://github.com/testing-library/user-event/issues/497) by default already contains an act wrapper. Try removing act and editing this test.

Reply:
Done